sdk/tables/Azure.Data.Tables/src/TableClient.cs

```diff
@@ -249,16 +255,28 @@ public virtual async Task<Response> InsertOrReplaceAsync(IDictionary<string, obj
                 throw new ArgumentException(TableConstants.ExceptionMessages.MissingRowKey, nameof(entity));
             }
 
-            using DiagnosticScope scope = _diagnostics.CreateScope($"{nameof(TableClient)}.{nameof(InsertOrReplace)}");
+            using DiagnosticScope scope = _diagnostics.CreateScope($"{nameof(TableClient)}.{nameof(UpsertEntity)}");
             scope.Start();
             try
             {
-                return await _tableOperations.UpdateEntityAsync(_table,
+                if (mode == UpdateMode.Replace)
+                {
+                    return await _tableOperations.UpdateEntityAsync(_table,
+                                                            partitionKey as string,
+                                                            rowKey as string,
+                                                            tableEntityProperties: entity.ToOdataAnnotatedDictionary(),
+                                                            queryOptions: new QueryOptions() { Format = _format },
+                                                            cancellationToken: cancellationToken).ConfigureAwait(false);
+                }
+                else
```

Comment:
This is a super nitpicky suggestion, but to be extra safe it's great to add an `else if (mode == UpdateMode.Merge)` and then add a final `else { throw new ArgumentException($"Unexpected value for {nameof(mode)}: {mode}");` in case someone adds new `UpdateMode` values but forgets to change this code.